### PR TITLE
Add recursive and type options (raw/docker) to spec and summary endpoints.

### PR DESCRIPTION
**Pull request by rjnagal**
spec, stats, and summary now all support the same options (except count,
which is only parsed for stats).

### CODE REVIEW

**Comment by vmarmol** [comment]
btw:

```
$ ./deploy/check_gofmt.sh .
The following files are not properly formatted:
api/versions.go
```

File: manager/manager.go [hunk 1]
```diff
@@ -66,8 +66,11 @@ type Manager interface {
 	// Gets spec for a container.
 	GetContainerSpec(containerName string) (v2.ContainerSpec, error)
 
-	// Get derived stats for a container.
-	GetContainerDerivedStats(containerName string) (v2.DerivedStats, error)
+	// Get data for all requested containers based on the request options.
+	GetRequestedContainers(containerName string, options v2.RequestOptions) (map[string]*containerData, error)
```
Comment (vmarmol) [inline manager/manager.go]: containerData wasn't meant to be publicly exposed. If we do this route we should make it safe to do so (doesn't have to be in this PR). I'm just worried about its use in Kubelet :)

We can also replace most of the stuff here with a call to this and then getting some data.

Comment (rjnagal) [inline manager/manager.go]: We can go back to exposing  manager.GetDerivedStats() and manager.GetSpec() and make containerData private. We'll just need to move that logic from api/ to manager/

Let me do that in this PR.

Comment (vmarmol) [inline manager/manager.go]: I'm +1 on exposing a Container object that we can GetSpec() and GetStats() from.

**Comment by rjnagal** [comment]
PTAL. Moved all containerData references back to manager.

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**